tests/test_no_input_provider_speech.py

```
@@ -0,0 +1,219 @@
+    def __init__(self, messages):
+        self._messages = list(messages)
+
+    def __aiter__(self):
+        self._iterator = iter(self._messages)
+        return self
+
+    async def __anext__(self):
+        try:
+            return next(self._iterator)
+        except StopIteration:
+            raise StopAsyncIteration
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)